src/libs/Navigation/Navigation.js

```diff
@@ -74,10 +75,11 @@ function closeDrawer() {
 }
 
 /**
- * @param {Boolean} isSmallScreenWidth
+ * @param {Boolean} isSmallScreen
  * @returns {String}
  */
-function getDefaultDrawerState(isSmallScreenWidth) {
+function getDefaultDrawerState(isSmallScreen) {
+    isSmallScreenWidth = isSmallScreen;
```

Comment:
This getter having a side effect of setting `isSmallScreenWidth` feels too dirty. `navigate` calling `navigationRef.current.dispatch(CustomActions.pushDrawerRoute(route, isSmallScreenWidth));` depends on `getDefaultDrawerState` having been called before.

Reply:
Are we okay using your approach to get the dimension?
```
const initialDimensions = Dimensions.get('window');
const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
```

Reply:
I couldn't find another way to get the screen size except using `Dimensions`.

Reply:
This is what I came with when I was testing, I didn't see any other way that didn't involve changing too many things (I must say I didn't tried much :P)

Thoughts @Santhosh-Sellavel ?

Reply:
Isn't there any API in Dimensions to listen to the changes? @mollfpr Explore the options.

Reply:
> Are we okay using your approach to get the dimension?
> 
> ```
> const initialDimensions = Dimensions.get('window');
> const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
> ```

This won't work.

Reply:
> This won't work.

Why?

Reply:
We measure only once here, then minimizing or expanding the window would not reflect correctly.

Reply:
I may be missing something, but if that code is inside `pushDrawerRoute`, wouldn't it be up to date when that function is called? `pushDrawerRoute` doesn't get called when minimizing or expanding the window, does it?

Reply:
> Yep, that should do.

@aldo-expensify I already confirmed it would work.

Reply:
> > Yep, that should do.
> 
> @aldo-expensify I already confirmed it would work.

The version with the listener?

Reply:
> Are we okay using your approach to get the dimension?
> 
> ```
> const initialDimensions = Dimensions.get('window');
> const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
> ```

Sorry for the confusion, I thought we are determining once. This way itself will work if it is determined every time.
@aldo-expensify